iceoryx_dust/container/include/iox/detail/fixed_position_container.inl

```diff
@@ -94,73 +82,79 @@ FixedPositionContainer<T, CAPACITY>::operator=(FixedPositionContainer&& rhs) noe
 {
     if (this != &rhs)
     {
-        copy_and_move_impl(std::move(rhs));
-
-        // clear rhs
-        rhs.clear();
+        copy_and_move_impl<detail::SpecialCreationOperations::MoveAssignment>(std::move(rhs));
     }
     return *this;
 }
 
 template <typename T, uint64_t CAPACITY>
-template <typename RhsType>
+template <detail::SpecialCreationOperations Opt, typename RhsType>
 inline void FixedPositionContainer<T, CAPACITY>::copy_and_move_impl(RhsType&& rhs) noexcept
 {
-    // we already make sure rhs is always passed by std::move() for move case, therefore
-    // the result of "decltype(rhs)" is same as "decltype(std::forward<RhsType>(rhs))"
-    static_assert(std::is_rvalue_reference<decltype(rhs)>::value
-                      || (std::is_lvalue_reference<decltype(rhs)>::value
-                          && std::is_const<std::remove_reference_t<decltype(rhs)>>::value),
-                  "RhsType must be const lvalue reference or rvalue reference");
+    static_assert(
+        std::is_rvalue_reference<decltype(rhs)>::value
+            || (std::is_lvalue_reference_v<decltype(rhs)> && std::is_const_v<std::remove_reference_t<decltype(rhs)>>),
+        "RhsType must be const lvalue reference or rvalue reference");
+
+    // alias helper struct
+    using Helper = detail::SpecialCreationHelper<Opt>;
 
-    constexpr bool is_move = std::is_rvalue_reference<decltype(rhs)>::value;
+    constexpr bool is_ctor = Helper::is_ctor();
+    constexpr bool is_move = Helper::is_move();
+
+    if constexpr (is_ctor)
+    {
+        for (IndexType i = 0; i < CAPACITY; ++i)
+        {
+            m_status[i] = SlotStatus::FREE;
+        }
+    }
 
     IndexType i{Index::FIRST};
     auto rhs_it = (std::forward<RhsType>(rhs)).begin();
 
-    // transfer src data to destination
     for (; rhs_it.to_index() != Index::INVALID; ++i, ++rhs_it)
     {
         if (m_status[i] == SlotStatus::USED)
         {
-#if __cplusplus >= 201703L
+            // When the slot is in the 'USED' state, it is safe to proceed with either construction (ctor) or assignment
+            // operation. Therefore, creation can be carried out according to the option specified by Opt.
+
+            // @todo iox-#2052: moveCtor will failed if using move_or_copy_it. Twice time moveCtor will be called.
+            // Helper::create(m_data[i], Helper::move_or_copy_it(rhs_it));
```

Comment:
I'm not sure I understand the comment :sweat_smile: 

Is this something temporal which will be fixed in this PR or is it just something which could replace the lines below?

Reply:
Ah, I should have specified that it's the `move_or_copy_it` issue. My intention was to replace the code from lines 126 to 133 with that on line 124, but I was unsuccessful. In the case of `movector`, it is called once within `move_or_copy_it`, which is not our desired outcome. Therefore, I'm still searching for a solution to this problem. You can view an example [here](https://godbolt.org/z/3xcPraPTh). If you have any suggestions, please feel free to share them with me.

Reply:
Okay. Would you move the commented code out before the merge if you don't find a solution. You could for example add a section to the description of #2052 and add the code there. Something like `## Evaluate foo` with an explanation or create a new issue ... might even be the issue to move the helper files to `iceoryx_hoofs/design`

Reply:
Ok.